README.md
# radiolog

[![radiolog screen shot](/doc/radiolog_screenshot.png)](https://www.youtube.com/watch?v=pTk-0i6uYUQ)

We've put together the 28-minute screencast video above, covering the key features.

Click any of the quick links in the 'Key Features' section below to start watching at that section, or at that section's demo.

---
[![radiolog fleetsync_screen shot](/doc/radiolog_fleetsync_screenshot.png)](https://youtu.be/Q786OtFgZDc)

November 2017: here's a new 8-minute screencast video on the FleetSync features of Radiolog, including FleetSync muting and selective device filtering.

---
[![radiolog non_fleetsync_screen shot](/doc/radiolog_non_fs_video2.png)](https://youtu.be/J_sFsYn4sbY)

January 2018: here's a new 4-minute summary screencast video on using Radiolog without Fleetsync.

---

Nevada County Sheriff's Search and Rescue (ncssar-nc.org) has been using this radio log tool on every callout since early 2015.

This open-source project is mainly intended for use by other Search and Rescue teams.  While it's not currently (Nov. 10 2016) in an easily installable state, one of our biggest issues will be to make a clean installer.  Until then, please contact the author if your team is interested in using this software.

See the latest installation procedure [here](INSTALL.md).

See guidelines for contributing to this project [here](CONTRIBUTING.md).

The general idea:

General Radio Log management
- Intended to be user-friendly and intuitive, with no training necessary to make use of key features
- Most features are automated
- Less overhead = Reduced Fiddling
- Fewer options = Reduced Fiddling
- Reduced Fiddling = more effective Radio Operator

Key Features

- [Introduction](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=0m0s)

- [Starting the program](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=1m27s)

- [Team tabs](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=1m59s)   ([demo](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=4m9s))

- [Welfare checks](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=4m43s)   ([demo](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=6m26s))

- [Team status](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=7m0s)   ([demo](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=8m41s))

- [Clue management](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=10m56s)    ([demo](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=15m5s))

- ['Subject located' button](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=16m15s)    ([demo](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=16m54s))

- [Kenwood FleetSync interface](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=18m2s)    ([SARSoft Locators demo](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=20m53s))

- [Message stack](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=21m58s)    ([demo](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=24m14s))

- [Amending messages](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=24m50s)

- [Multi-operational-period support](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=26m13s)

- [Printing](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=17m36s)

- [Automatic file management](https://www.youtube.com/watch?v=pTk-0i6uYUQ&t=28m11s)


Overview slideshow (17 slides describing the features in a bit more detail): [overview PDF](/doc/radiolog_overview.pdf)

FleetSync interface details are spelled out [here](/doc/radiolog_fleetsync_details.pdf).

Please contact the author if other documentation would be helpful.


doc_technical/CHANGE_LOG.adoc
= RadioLog Change Log


== Version 2.0.2 Release

Tagging this code as a "release" prior to Craig merging in some heavy-duty
pull requests -- just in case something goes horriby wrong and we need to
revert to this point and start a new branch.

[width="100%",cols="",options="header"]
|===
| DATE       | AUTHOR | NOTES
|   7-21-20  | TMG    | fix #428: LOCATED A CLUE error during amended message
                        leaves 'zombie' clue dialog preventing entry closure
|   7-21-20  | TMG    | fix #429: handle decode error after hot-plug

|===


== Version 2.0.0 "release" (not actually tagged as such)

This is the big switch from an external dependency of `GISInternals` to
the Python module `pyproj`.

[width="100%",cols="",options="header"]
|===
| DATE       | AUTHOR | NOTES
|   6-24-20  | TMG    | fix #48: use pyproj instead of GISInternals for conversion
|===


== Prior Changes

Here is the change log that was manually maintained through 6-15-2020:

[width="100%",cols="",options="header"]
|===
| DATE       | AUTHOR | NOTES
| 2-22-15    | TMG    | First version installed and working on NCSSAR Computer 2
| 4-9-15     | TMG    | Feature-complete release candidate
| 4-10-15    | TMG    | enable reading from a second com port (two radios)
| 4-28-15    | TMG    | First release; initial commit to git
| 5-11-15    | TMG    | fix bug 10: don't clear team timer unless the message
                        is 'FROM' with non-blank message text; change more
                        app-wide stylesheet font sizes in fontsChanged, and
                        change in children to override as needed
| 11-27-16   | TMG    | fix 307 (help window flashing colors are bouncing); also
                        verified no ill effects in team tabs
| 12-1-16    | TMG    | fix 268 (throb crash on oldest item in non-empty stack)
| 12-1-16    | TMG    | add -nosend option to disable sending of GET requests,
                        to avoid lag when network is not present
| 12-10-16   | TMG    | fix 267 (filename space handler) - remove spaces from
                        incident name for purposes of filenames
| 12-10-16   | TMG    | fix 25 (print dialog button wording should change when
                        called from main window exit button)
| 12-10-16   | TMG    | fix 306 (attached callsigns)
|  1-17-17   | TMG    | fix 41 (USB hot-plug / hot-unplug)
|  2-26-17   | TMG    | fix 311 (attached callsign bug)
|  2-27-17   | TMG    | fix 314 (NED focus / two-blinking-cursors)
|  2-27-17   | TMG    | fix 315 (ctrl-Z crash on empty list) for NED and for clue report
|  2-27-17   | TMG    | fix 312 (prevent orphaned clue/subject dialogs);
                        fix 317 (crash on cancel of cancel for clue/subject dialogs);
                        this involved changing dialog cancel buttons to just call
                        close() instead of reject()
|  2-28-17   | TMG    | fix 316,318,320 (add dialog open/cancel radio log entries)
                        and extend closeEvent functionality (above) to
                        nonRadioClueDialog and changeCallsignDialog
|   4-7-17   | TMG    | stopgap for 310 - disable attached callsign handling for now
|  4-11-17   | TMG    | fix 322 (restore crashes - file not found) - give a message
                        and return gracefully if the file specified in the rc file
                        does not exist, and, take steps to make sure the correct
                        filename is saved to the rc file at the correct times
|  4-15-17   | TMG    | fix 323 (load dialog should only show non-fleetsync and
                        non-clueLog .csv files)
|  4-26-17   | TMG    | fix 326 (zero-left-padded tabs when callsigns are digits-only)
|  4-29-17   | TMG    | fix 34 (fleetsync mute)
|  5-2-17    | TMG    | fix 325 (cancel-confirm bypass if message is blank)
|  5-13-17   | TMG    | fix 338 (esc key in clue/subject dialogs closes w/out confirm):
                        add keyPressEvents to ignore the esc key; note that the
                        Qt docs say that .reject() is called by the esc key but
                        that has other repercussions in this case; also serves
                        as interim fix for #337 (crash due to the above) until
                        a strong parent-child dialog relationship can be established
                        see http://stackoverflow.com/questions/43956587)
|  5-13-17   | TMG    | fix #333 (crash on throb after widget has been closed)
|  5-13-17   | TMG    | further fix on #333: don't try to stop the timer if it
                        does not exist, i.e. if not currently mid-throb; caused
                        similar crash to #333 during auto-cleanup of delayed stack
|  5-19-17   | TMG    | move loadFlag settings closer to core load functionality
                        to at least partially address #340
|  5-21-17   | TMG    | fix #257, fix #260: 90+% lag reduction: do not do any sorting;
                        instead, calculate the correct index to insert a new row
                        during newEntry, and use beginInsertRows and endInsertRows
                        instead of layoutChanged; see notes in newEntry function
|  6-15-17   | TMG    | fix #336 by simply ignoring keyPressEvents that happen
                        before newEntryWidget is responsive, and get fielded by
                        MyWindows instead; would be nice to find a better long-term
                        solution; see https://stackoverflow.com/questions/44148992
                        and see notes inline below
|  7-1-17    | TMG    | fix #342 (focus-follows-mouse freeze); 'fix' #332 (freeze
                        due to modal dialogs displayed underneath other windows)
                        by doing full audit and recode and test of all QMessageBox calls
|  7-1-17    | TMG    | fix #343 (crash on print clue log when there are no clues):
                        show an error message when the print button is clicked if

clueDialog_ui.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'clueDialog.ui'
#
# Created by: PyQt5 UI code generator 5.15.0
#
# WARNING: Any manual changes made to this file will be lost when pyuic5 is
# run again.  Do not edit this file unless you know what you are doing.


from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_clueDialog(object):
    def setupUi(self, clueDialog):
        clueDialog.setObjectName("clueDialog")
        clueDialog.resize(728, 561)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        clueDialog.setFont(font)
        self.label_4 = QtWidgets.QLabel(clueDialog)
        self.label_4.setGeometry(QtCore.QRect(10, 370, 141, 37))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        self.label_4.setFont(font)
        self.label_4.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_4.setObjectName("label_4")
        self.buttonBox = QtWidgets.QDialogButtonBox(clueDialog)
        self.buttonBox.setGeometry(QtCore.QRect(550, 440, 161, 51))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(14)
        self.buttonBox.setFont(font)
        self.buttonBox.setOrientation(QtCore.Qt.Horizontal)
        self.buttonBox.setStandardButtons(QtWidgets.QDialogButtonBox.Cancel|QtWidgets.QDialogButtonBox.Ok)
        self.buttonBox.setObjectName("buttonBox")
        self.label_3 = QtWidgets.QLabel(clueDialog)
        self.label_3.setGeometry(QtCore.QRect(20, 310, 131, 37))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        self.label_3.setFont(font)
        self.label_3.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_3.setObjectName("label_3")
        self.label = QtWidgets.QLabel(clueDialog)
        self.label.setGeometry(QtCore.QRect(0, 170, 151, 37))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        self.label.setFont(font)
        self.label.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label.setObjectName("label")
        self.instructionsField = QtWidgets.QLineEdit(clueDialog)
        self.instructionsField.setGeometry(QtCore.QRect(160, 370, 551, 40))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        self.instructionsField.setFont(font)
        self.instructionsField.setObjectName("instructionsField")
        self.locationField = QtWidgets.QLineEdit(clueDialog)
        self.locationField.setGeometry(QtCore.QRect(160, 310, 551, 39))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        self.locationField.setFont(font)
        self.locationField.setObjectName("locationField")
        self.label_2 = QtWidgets.QLabel(clueDialog)
        self.label_2.setGeometry(QtCore.QRect(270, 20, 191, 41))
        font = QtGui.QFont()
        font.setPointSize(20)
        self.label_2.setFont(font)
        self.label_2.setAlignment(QtCore.Qt.AlignCenter)
        self.label_2.setObjectName("label_2")
        self.clueNumberField = QtWidgets.QLineEdit(clueDialog)
        self.clueNumberField.setGeometry(QtCore.QRect(640, 20, 61, 31))
        font = QtGui.QFont()
        font.setPointSize(14)
        self.clueNumberField.setFont(font)
        self.clueNumberField.setObjectName("clueNumberField")
        self.label_6 = QtWidgets.QLabel(clueDialog)
        self.label_6.setGeometry(QtCore.QRect(560, 20, 71, 31))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(12)
        self.label_6.setFont(font)
        self.label_6.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.label_6.setObjectName("label_6")
        self.groupBox = QtWidgets.QGroupBox(clueDialog)
        self.groupBox.setGeometry(QtCore.QRect(30, 70, 661, 92))
        self.groupBox.setTitle("")
        self.groupBox.setObjectName("groupBox")
        self.callsignField = QtWidgets.QLineEdit(self.groupBox)
        self.callsignField.setEnabled(False)
        self.callsignField.setGeometry(QtCore.QRect(240, 20, 161, 31))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(12)
        self.callsignField.setFont(font)
        self.callsignField.setFocusPolicy(QtCore.Qt.NoFocus)
        self.callsignField.setText("")
        self.callsignField.setObjectName("callsignField")
        self.label_5 = QtWidgets.QLabel(self.groupBox)
        self.label_5.setEnabled(False)
        self.label_5.setGeometry(QtCore.QRect(240, 40, 161, 41))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(9)
        self.label_5.setFont(font)
        self.label_5.setAlignment(QtCore.Qt.AlignCenter)
        self.label_5.setObjectName("label_5")
        self.label_8 = QtWidgets.QLabel(self.groupBox)
        self.label_8.setEnabled(False)
        self.label_8.setGeometry(QtCore.QRect(424, 59, 221, 41))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(9)
        self.label_8.setFont(font)
        self.label_8.setAlignment(QtCore.Qt.AlignCenter)
        self.label_8.setObjectName("label_8")
        self.label_7 = QtWidgets.QLabel(self.groupBox)
        self.label_7.setEnabled(False)
        self.label_7.setGeometry(QtCore.QRect(130, 40, 91, 41))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(9)
        self.label_7.setFont(font)
        self.label_7.setAlignment(QtCore.Qt.AlignCenter)
        self.label_7.setObjectName("label_7")
        self.timeField = QtWidgets.QLineEdit(self.groupBox)
        self.timeField.setEnabled(False)
        self.timeField.setGeometry(QtCore.QRect(130, 20, 91, 31))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(12)
        self.timeField.setFont(font)
        self.timeField.setFocusPolicy(QtCore.Qt.NoFocus)
        self.timeField.setText("")
        self.timeField.setObjectName("timeField")
        self.dateField = QtWidgets.QLineEdit(self.groupBox)
        self.dateField.setEnabled(False)
        self.dateField.setGeometry(QtCore.QRect(20, 20, 91, 31))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(12)
        self.dateField.setFont(font)
        self.dateField.setFocusPolicy(QtCore.Qt.NoFocus)
        self.dateField.setText("")
        self.dateField.setObjectName("dateField")
        self.label_9 = QtWidgets.QLabel(self.groupBox)
        self.label_9.setEnabled(False)
        self.label_9.setGeometry(QtCore.QRect(20, 40, 91, 41))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(9)
        self.label_9.setFont(font)
        self.label_9.setAlignment(QtCore.Qt.AlignCenter)
        self.label_9.setObjectName("label_9")
        self.radioLocField = QtWidgets.QTextEdit(self.groupBox)
        self.radioLocField.setEnabled(False)
        self.radioLocField.setGeometry(QtCore.QRect(429, 6, 211, 63))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.radioLocField.setFont(font)
        self.radioLocField.setFocusPolicy(QtCore.Qt.NoFocus)
        self.radioLocField.setObjectName("radioLocField")
        self.clueQuickTextButton1 = QtWidgets.QPushButton(clueDialog)
        self.clueQuickTextButton1.setGeometry(QtCore.QRect(20, 430, 161, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.clueQuickTextButton1.setFont(font)
        self.clueQuickTextButton1.setFocusPolicy(QtCore.Qt.NoFocus)
        self.clueQuickTextButton1.setObjectName("clueQuickTextButton1")
        self.clueQuickTextButton3 = QtWidgets.QPushButton(clueDialog)
        self.clueQuickTextButton3.setGeometry(QtCore.QRect(400, 430, 131, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.clueQuickTextButton3.setFont(font)
        self.clueQuickTextButton3.setFocusPolicy(QtCore.Qt.NoFocus)
        self.clueQuickTextButton3.setObjectName("clueQuickTextButton3")
        self.clueQuickTextButton4 = QtWidgets.QPushButton(clueDialog)
        self.clueQuickTextButton4.setGeometry(QtCore.QRect(20, 470, 161, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.clueQuickTextButton4.setFont(font)
        self.clueQuickTextButton4.setFocusPolicy(QtCore.Qt.NoFocus)
        self.clueQuickTextButton4.setObjectName("clueQuickTextButton4")
        self.clueQuickTextButton2 = QtWidgets.QPushButton(clueDialog)
        self.clueQuickTextButton2.setGeometry(QtCore.QRect(200, 430, 181, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.clueQuickTextButton2.setFont(font)
        self.clueQuickTextButton2.setFocusPolicy(QtCore.Qt.NoFocus)
        self.clueQuickTextButton2.setObjectName("clueQuickTextButton2")
        self.clueQuickTextUndoButton = QtWidgets.QPushButton(clueDialog)
        self.clueQuickTextUndoButton.setGeometry(QtCore.QRect(20, 520, 121, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.clueQuickTextUndoButton.setFont(font)
        self.clueQuickTextUndoButton.setFocusPolicy(QtCore.Qt.NoFocus)
        self.clueQuickTextUndoButton.setObjectName("clueQuickTextUndoButton")
        self.clueReportPrintCheckBox = QtWidgets.QCheckBox(clueDialog)
        self.clueReportPrintCheckBox.setGeometry(QtCore.QRect(180, 520, 381, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.clueReportPrintCheckBox.setFont(font)
        self.clueReportPrintCheckBox.setChecked(True)
        self.clueReportPrintCheckBox.setObjectName("clueReportPrintCheckBox")
        self.clueQuickTextButton5 = QtWidgets.QPushButton(clueDialog)
        self.clueQuickTextButton5.setGeometry(QtCore.QRect(200, 470, 181, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.clueQuickTextButton5.setFont(font)
        self.clueQuickTextButton5.setFocusPolicy(QtCore.Qt.NoFocus)
        self.clueQuickTextButton5.setObjectName("clueQuickTextButton5")
        self.descriptionField = QtWidgets.QPlainTextEdit(clueDialog)
        self.descriptionField.setGeometry(QtCore.QRect(160, 170, 551, 121))
        self.descriptionField.viewport().setProperty("cursor", QtGui.QCursor(QtCore.Qt.IBeamCursor))
        self.descriptionField.setMouseTracking(True)
        self.descriptionField.setFocusPolicy(QtCore.Qt.StrongFocus)
        self.descriptionField.setStyleSheet("font: 14pt \"Segoe UI\";")
        self.descriptionField.setInputMethodHints(QtCore.Qt.ImhNone)
        self.descriptionField.setTabChangesFocus(True)
        self.descriptionField.setObjectName("descriptionField")
        self.clueQuickTextButton6 = QtWidgets.QPushButton(clueDialog)
        self.clueQuickTextButton6.setGeometry(QtCore.QRect(400, 470, 131, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.clueQuickTextButton6.setFont(font)
        self.clueQuickTextButton6.setFocusPolicy(QtCore.Qt.NoFocus)
        self.clueQuickTextButton6.setObjectName("clueQuickTextButton6")
        self.line = QtWidgets.QFrame(clueDialog)
        self.line.setGeometry(QtCore.QRect(100, 500, 531, 20))
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.groupBox.raise_()
        self.label_4.raise_()
        self.buttonBox.raise_()
        self.label_3.raise_()
        self.label.raise_()
        self.instructionsField.raise_()
        self.locationField.raise_()
        self.label_2.raise_()
        self.clueNumberField.raise_()
        self.label_6.raise_()
        self.clueQuickTextButton1.raise_()
        self.clueQuickTextButton3.raise_()
        self.clueQuickTextButton4.raise_()
        self.clueQuickTextButton2.raise_()
        self.clueQuickTextUndoButton.raise_()
        self.clueReportPrintCheckBox.raise_()
        self.clueQuickTextButton5.raise_()
        self.descriptionField.raise_()
        self.clueQuickTextButton6.raise_()
        self.line.raise_()

        self.retranslateUi(clueDialog)
        self.buttonBox.accepted.connect(clueDialog.accept)
        self.buttonBox.rejected.connect(clueDialog.close)
        self.clueQuickTextButton1.clicked.connect(clueDialog.clueQuickTextAction)
        self.clueQuickTextButton2.clicked.connect(clueDialog.clueQuickTextAction)
        self.clueQuickTextButton3.clicked.connect(clueDialog.clueQuickTextAction)
        self.clueQuickTextButton4.clicked.connect(clueDialog.clueQuickTextAction)
        self.clueQuickTextUndoButton.clicked.connect(clueDialog.clueQuickTextUndo)
        self.clueQuickTextButton5.clicked.connect(clueDialog.clueQuickTextAction)
        self.clueQuickTextButton6.clicked.connect(clueDialog.clueQuickTextAction)
        QtCore.QMetaObject.connectSlotsByName(clueDialog)
        clueDialog.setTabOrder(self.clueNumberField, self.descriptionField)
        clueDialog.setTabOrder(self.descriptionField, self.locationField)
        clueDialog.setTabOrder(self.locationField, self.instructionsField)
        clueDialog.setTabOrder(self.instructionsField, self.clueReportPrintCheckBox)

    def retranslateUi(self, clueDialog):
        _translate = QtCore.QCoreApplication.translate
        clueDialog.setWindowTitle(_translate("clueDialog", "Clue Report"))
        self.label_4.setText(_translate("clueDialog", "Instructions"))
        self.label_3.setText(_translate("clueDialog", "Location"))
        self.label.setText(_translate("clueDialog", "Description"))
        self.label_2.setText(_translate("clueDialog", "Clue Report"))
        self.label_6.setText(_translate("clueDialog", "Clue #"))
        self.label_5.setText(_translate("clueDialog", "REPORTED BY"))
        self.label_8.setText(_translate("clueDialog", "RADIO LOCATION"))
        self.label_7.setText(_translate("clueDialog", "TIME"))
        self.label_9.setText(_translate("clueDialog", "DATE"))
        self.clueQuickTextButton1.setText(_translate("clueDialog", "COLLECT  [F1]"))
        self.clueQuickTextButton1.setShortcut(_translate("clueDialog", "F1"))
        self.clueQuickTextButton3.setText(_translate("clueDialog", "DISREGARD  [F3]"))
        self.clueQuickTextButton3.setShortcut(_translate("clueDialog", "F3"))
        self.clueQuickTextButton4.setText(_translate("clueDialog", "HOLD POSITION  [F4]"))
        self.clueQuickTextButton4.setShortcut(_translate("clueDialog", "F4"))
        self.clueQuickTextButton2.setText(_translate("clueDialog", "MARK && LEAVE  [F2]"))
        self.clueQuickTextButton2.setShortcut(_translate("clueDialog", "F2"))
        self.clueQuickTextUndoButton.setText(_translate("clueDialog", "UNDO  [Ctrl+Z]"))
        self.clueQuickTextUndoButton.setShortcut(_translate("clueDialog", "Ctrl+Z"))
        self.clueReportPrintCheckBox.setText(_translate("clueDialog", "Print a Clue Report Form as soon as this clue is saved"))
        self.clueQuickTextButton5.setText(_translate("clueDialog", "PROTECT THE CLUE  [F5]"))
        self.clueQuickTextButton5.setShortcut(_translate("clueDialog", "F5"))
        self.clueQuickTextButton6.setText(_translate("clueDialog", "STANDBY [F6]"))
        self.clueQuickTextButton6.setShortcut(_translate("clueDialog", "F6"))
